### PR TITLE
[FIX] dice loss for 2 classes + contour weighting

### PR DESCRIPTION
**Pull request by fantes**
- caffe dice connexion with prototxt was still done as in an old version of dice layer, fix is to use general case (ie >2 classes)
- dice contour weighting has a bug (normalization ??) disabling it for now

### CODE REVIEW

File: src/backends/caffe/caffelib.cc [hunk 1]
```diff
@@ -4010,18 +3946,16 @@ void CaffeLib<TInputConnectorStrategy,TOutputConnectorStrategy,TMLModel>::update
                     this->_logger->warn("dice contour amplitude unrecognized, float expected");
                   }
               }
+			this->_logger->warn("dice contour is not working (2020 - 05 - 19) disabling it");
+			contours = caffe::DiceCoefLossParameter::NO;
+
```
Comment (beniz) [inline src/backends/caffe/caffelib.cc:3951]: Let's fail the creation / job if contour is selected, no ?

**Comment by beniz** [comment]
Does this work for 2 classes then ?

**Comment by fantes** [comment]
> Does this work for 2 classes then ?

yes :)

File: src/backends/caffe/caffelib.cc (hunk 1 above)
Comment (fantes) [reply src/backends/caffe/caffelib.cc:3952]: done

**Pull request merged**